python/katana/property_graph.pyx.jinja

```diff
@@ -28,27 +31,32 @@ cdef class PropertyGraph:
     """
     A property graph loaded into memory.
     """
-    def __init__(self, path, node_properties=None, edge_properties=None):
+    def __init__(self, path, partition_to_load=None, node_properties=None, edge_properties=None):
         """
-        __init__(self, path, node_properties=None, edge_properties=None)
+        __init__(self, path, partition_to_load=None, node_properties=None, edge_properties=None)
 
         Load a property graph.
 
         :param path: the path or URL from which to load the graph. This support local paths or s3 URLs.
         :type path: str
+        :param partition_to_load: An unsigned integer corresponding to a partition number in the graph being loaded. If this is None (default), then the partition corresponding to this host's network id will be loaded.
         :param node_properties: A list of node property names to load into memory. If this is None (default), then all properties are loaded.
         :param edge_properties: A list of edge property names to load into memory. If this is None (default), then all properties are loaded.
         """
-        if node_properties is not None or edge_properties is not None:
-            if node_properties is None or edge_properties is None:
-                raise ValueError("If either node_properties or edge_properties are provided, both must be provided.")
-            self.underlying = handle_result_value(
-                _PropertyGraph.MakeWithProperties(bytes(path, "utf-8"),
-                                                     _convert_string_list(node_properties),
-                                                     _convert_string_list(edge_properties)))
-        else:
-            self.underlying = handle_result_value(
-                _PropertyGraph.Make(bytes(path, "utf-8")))
+        cdef RDGLoadOptions opts
+        cdef vector[string] node_props
+        cdef vector[string] edge_props
+        cdef uint32_t part_to_load
+        if partition_to_load is not None:
+            part_to_load = partition_to_load
+            opts.partition_id_to_load = part_to_load
```

Comment:
You don't need the intermediate variable for things assigned to a value type field. The only reason for the intermediate is to have something to take a reference to that will be in scope for the length of the function.

Reply:
This field is not a value type, its type is `optional[uint32_t]`. And `optional` is always constructed from a reference. I will add a comment clarifying this.